### PR TITLE
PR: [Concept Entry] Discrete Math Number Bases

### PR DESCRIPTION
**Pull request by sreeharsha-rav**
### Description

- Created a concept entry for discrete math content. Added `number-bases/number-bases.md`.
- Closes Issue #2750 

### Type of Change

- Added definition of number bases
- Described types of number bases
- Briefly introduced how to convert between number bases
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:28]:
```suggestion
### Binary

[Binary](https://www.codecademy.com/resources/docs/general/binary) is a number system that uses a base of 2. A number in binary can only be represented by `0`'s and `1`'s. In computer languages, the notation for representing binary numbers is leading `0b`. For example, `0b1010` is a binary number.

```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:31]:
```suggestion

```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:33]:
```suggestion
### Octal

Octal is a number system that uses a base of 8. A number in octal can only be represented by eight digits `0-7`. In computer languages, the notation for representing octal numbers is leading `0o`. For example, `0o0176` is an octal number.
```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:36]:
```suggestion
### Decimal

Decimal is the most common number system that uses a base of 10. A number in decimal contains digits `0-9`. In computer languages, by default there is no notation required to represent decimal numbers. For example, `9801` is a decimal number.
```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:41]:
```suggestion
### Hexadecimal 

[Hexadecimal](https://www.codecademy.com/resources/docs/general/hexadecimal) is a number system that uses a base of 16. The first ten digits of a hexadecimal number contains digits `0-9` and the remaining six digits contains letters `A-F` (not case sensitive). In computer languages, the notation for representing hexadecimal numbers is leading `0x`. For example, `0xA63F` is a hexadecimal number.
```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:44]:
```suggestion
## Converting between Bases

Converting between number bases requires two key concepts: the place value system and the digit representation.

```
We avoid first- and second-person pronouns in Docs.

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:55]:
```suggestion
### Binary to Other Bases

- To convert binary (base-2) to decimal (base-10), multiply each digit by 2 raised to its respective power and sum them up. For example, `0b1001` in binary is `1 * 2^3 + 0 * 2^2 + 0 * 2^1 + 1 * 2^0 = 9` in decimal.
- To convert binary to octal (base-8), group the binary digits into sets of three (from the right) and convert each group to its octal equivalent. For example, `0b001100` in binary is grouped as `001 100`, and the octal equivalent is `0o14`.
- To convert binary to hexadecimal (base-16), group the binary digits into sets of four (from the right) and convert each group to its hexadecimal equivalent. For example, `0b10011111` in binary is grouped as `1001 1111`, and the hexadecimal equivalent is `0x9F`.
```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:62]:
```suggestion
### Octal to Other Bases

- To convert octal (base-8) to decimal, multiply each digit by 8 raised to its respective power and sum them up. For example, `0o67` in octal is `6 * 8^1 + 7 * 8^0 = 55` in decimal.
- To convert octal to binary, convert each octal digit to its 3-bit binary equivalent. For example, `0o67` in octal is `110 111` in binary.
- To convert octal to hexadecimal, first convert the octal number to binary and then group the binary digits into sets of four and convert each group to its hexadecimal equivalent.
```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:64]:
```suggestion
### Decimal to Other Bases

```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:65]:
```suggestion
- To convert decimal (base-10) to binary, repeatedly divide the decimal number by `2`, keeping track of the remainders. The binary equivalent is obtained by reading the remainders in reverse order. 

```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:78]:
```suggestion
- To convert decimal to octal, repeatedly divide the decimal number by `8`, keeping track of the remainders. The octal equivalent is obtained by reading the remainders in reverse order.
- To convert decimal to hexadecimal, repeatedly divide the decimal number by `16`, keeping track of the remainders. The hexadecimal equivalent is obtained by reading the remainders in reverse order, with `10` represented as `A`, `11` as `B`, and so on.
```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:85]:
```suggestion
### Hexadecimal to Other Bases

- To convert hexadecimal (base-16) to decimal, multiply each digit by 16 raised to its respective power and sum them up. For example, `0x3F` in hexadecimal is `3 * 16^1 + 15 * 16^0 = 63` in decimal.
- To convert hexadecimal to binary, convert each hexadecimal digit to its 4-bit binary equivalent. For example, `0x3F` in hexadecimal is `0011 1111` in binary.
- To convert hexadecimal to octal, first convert the hexadecimal number to binary and then group the binary digits into sets of three and convert each group to its octal equivalent.
```

File: content/discrete-math/concepts/number-bases/number-bases.md
Comment (SSwiniarski) [inline content/discrete-math/concepts/number-bases/number-bases.md:25]:
```suggestion
## Types of Bases

A base can be any whole number greater than 1. The four most commonly used number bases in a computer system are:

  - Binary (Base 2)
  - Octal (Base 8)
  - Decimal (Base 10)
  - Hexadecimal (Base 16)
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@sreeharsha-rav I dis a first review and left a number of comments.

**Review by SSwiniarski — APPROVED** [review]
@sreeharsha-rav I did one minor edit and this is good for a second review.

**Comment by sreeharsha-rav** [comment]
Should I close the pull request? or is it still awaiting approval, I am confused.

**Review by KTom101 — CHANGES_REQUESTED** [review]
@sreeharsa-rav I've added a few changes.

**Review by KTom101 — APPROVED** [review]
@sreeharsha-rav I added a few more changes directly and ran `npm run format:all` to fix the failing [checks](https://www.codecademy.com/resources/docs/contribution-guide/getting-set-up-on-your-computer#:~:text=continuous%20integration%20(CI)%20workflows%20in%20place%20to%20test%20changes).

**Pull request merged**
